internal/backends/postgresql/query_iterator.go

```diff
@@ -89,13 +91,7 @@ func (iter *queryIterator) Next() (struct{}, *types.Document, error) {
 		return unused, nil, lazyerrors.Error(err)
 	}
 
-	var b []byte
-	if err := iter.rows.Scan(&b); err != nil {
-		iter.close()
-		return unused, nil, lazyerrors.Error(err)
-	}
-
-	doc, err := sjson.Unmarshal(b)
+	doc, err := iter.s.Scan(iter.rows)
```

Comment:
I don't think we should introduce the whole separate entity with an interface and implementations there.

Instead, we could use `iter.rows.FieldDescriptions()` once to get information about columns and how to scan rows

Reply:
It's simple for sqlite too, just exposed additional function `Columns` from sql